src/index.js

```diff
@@ -1,172 +1,105 @@
 
-    // fail last
-    if (failedUrls.length) {
-      core.setFailed(
-        `Performance budget fails for ${failedUrls.length} URL${failedUrls.length === 1 ? '' : 's'}` +
-          ` (${failedUrls.join(', ')})`
-      )
+    status = await runChildCommand('upload', args)
+
+    if (status !== 0) {
+      throw new Error(`LHCI 'upload' has encountered a problem.`)
     }
-  } finally {
-    if (chrome) await chrome.kill()
+    core.endGroup() // Uploading
   }
 }
 
 // run `main()`
-
 main()
   .catch(
     /** @param {Error} err */ err => {
       core.setFailed(err.message)
-      process.exit(1)
     }
   )
   .then(() => {
     console.log(`done in ${process.uptime()}s`)
-    process.exit()
   })
 
 /**
- * Get urls from `url` or `urls`
+ * Run a child command synchronously.
  *
- * @return {string[]}
- */
-
-function getUrls() {
-  const url = core.getInput('url')
-  if (url) return [url]
-  const urls = core.getInput('urls')
-  return urls.split('\n').map(url => url.trim())
-}
-
-/** @return {object} */
-function getConfig() {
-  const configPath = core.getInput('configPath')
-  if (configPath) return require(join(process.cwd(), configPath))
-  return {
-    extends: 'lighthouse:default',
-    settings: {}
-  }
-}
-
-/** @return {string[] | null} */
-function getOnlyCategories() {
-  const onlyCategories = core.getInput('onlyCategories')
-  if (!onlyCategories) return null
-  return onlyCategories.split(',').map(category => category.trim())
-}
-
-/** @return {object | null} */
-function getBudgets() {
-  const budgetPath = core.getInput('budgetPath')
-  if (!budgetPath) return null
-  return JSON.parse(readFileSync(join(process.cwd(), budgetPath), 'utf8'))
-}
-
-/** @return {object | null} */
-function getExtraHeaders() {
-  const extraHeaders = core.getInput('extraHeaders')
-  if (!extraHeaders) return null
-  try {
-    return mapKeys(
-      JSON.parse(extraHeaders || '{}'),
-      /** @param {string} _val @param {string} key */ (_val, key) => key.toLowerCase()
-    )
-  } catch (err) {
-    console.error('Error at parsing extra headers:')
-    console.error(err)
-    return {}
-  }
-}
-
-/**
- * Parse flags: https://github.com/GoogleChrome/chrome-launcher/blob/master/docs/chrome-flags-for-tools.md
- * @return {string[]}
- */
-
-function getChromeFlags() {
-  const flags = ['--headless', '--disable-gpu', '--no-sandbox', '--no-zygote']
-  const chromeFlags = core.getInput('chromeFlags')
-  if (chromeFlags) flags.push(...chromeFlags.split(' '))
-  return flags
-}
-
-/**
- * Check if the performance budget exceed,
- * by looking at `sizeOverBudget` at `lhr.audits['performance-budget'].details.items`
- *
- * @param {object} lhr
- * @return {boolean}
+ * @param {'collect'|'assert'|'upload'} command
+ * @param {string[]} [args]
+ * @return {number}
  */
+function runChildCommand(command, args = []) {
+  const combinedArgs = [lhciCliPath, command, ...args]
+  const { status = -1 } = childProcess.spawnSync(process.argv[0], combinedArgs, {
```

Comment:
any reason to use `spawnSync` over `spawn`?

Reply:
`spawnSync` is easier to work with imo in this case. 

Since we really just care how the process ended, and we want to call it synchronously.